futurex_openedx_extensions/dashboard/details/learners.py

```diff
@@ -0,0 +1,81 @@
+"""Learners details collectors"""
+from __future__ import annotations
+
+from typing import List
+
+from common.djangoapps.student.models import CourseAccessRole, UserSignupSource
+from django.contrib.auth import get_user_model
+from django.db.models import Count, Exists, OuterRef, Q, Subquery
+from django.db.models.query import QuerySet
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+from futurex_openedx_extensions.helpers.tenants import get_course_org_filter_list, get_tenant_site
+
+
+def get_learners_queryset(tenant_ids: List, search_text: str = None) -> QuerySet:
```

Comment:
This query is a bit slow on production. Any idea how can we make it faster?

Reply:
I believe the issue is not this particular query. It's the other helper queries, you'll find a few methods with `TODO` to add cache. These will increase the performance significantly as far as I can see

Reply:
if caching didn't fix it, we might need to do something with `UserSignupSource`. We need it to record logins, not just signups. So we can use it to directly filter the user by site.

Currently, there are too many records for users with enrollments in a site, but without a record in `UserSignupSource`